### PR TITLE
[TSv4] ZkSync AA & Default Factory

### PR DESCRIPTION
**Pull request by 0xFirekeeper**
## PR-Codex overview
This PR introduces support for ZkSync Smart Wallet and Default Account Factory, updates SDK versions, and refactors smart wallet connectors.

### Detailed summary
- Added ZkSync Smart Wallet support
- Implemented Default Account Factory
- Updated SDK versions
- Refactored smart wallet connectors for ZkSync integration

> The following files were skipped due to too many changes: `legacy_packages/wallets/src/evm/connectors/smart-wallet/zk-connector.ts`, `legacy_packages/wallets/src/evm/connectors/smart-wallet/zk-wrapped-signer.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: legacy_packages/wallets/src/evm/connectors/smart-wallet/zk-wrapped-signer.ts [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    if (!transaction.chainId) {
+      throw new Error("ChainId is required to send a ZkSync transaction");
+    }
+
+    if (!this.provider) {
+      throw new Error("Provider is required to send a ZkSync transaction");
+    }
+
+    const address = await this.getAddress();
+    const gasLimit = (await this.provider.estimateGas(transaction)).mul(3);
```
Comment (joaquim-verges) [inline legacy_packages/wallets/src/evm/connectors/smart-wallet/zk-wrapped-signer.ts:126]: Populate transaction should already estimate under the hood I believe

File: legacy_packages/wallets/src/evm/connectors/smart-wallet/zk-wrapped-signer.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+      ).toNumber(),
+      paymaster: BigInt(pmDataResult.paymaster).toString(),
+      nonce: ethers.BigNumber.from(transaction.nonce).toNumber(),
+      value: ethers.BigNumber.from(transaction.value).toNumber(),
+      data: transaction.data || "0x",
+      factoryDeps: [],
+      paymasterInput: ethers.utils.arrayify(pmDataResult.paymasterInput),
+    };
+
+    console.log("eip712tx", eip712tx);
```
Comment (joaquim-verges) [inline legacy_packages/wallets/src/evm/connectors/smart-wallet/zk-wrapped-signer.ts:185]: Logs

File: legacy_packages/wallets/src/evm/wallets/smart-wallet.ts [hunk 3]
```diff
@@ -293,14 +297,26 @@ export class SmartWallet extends AbstractClientWallet<
...
     if (!this.connector) {
-      const { SmartWalletConnector } = await import(
-        "../connectors/smart-wallet"
-      );
-      this.connector = new SmartWalletConnector(
-        this.options as SmartWalletConfig,
-      );
+      if (
+        this.options?.chain === ZksyncSepoliaTestnet ||
+        this.options?.chain === Zksync
```
Comment (joaquim-verges) [inline legacy_packages/wallets/src/evm/wallets/smart-wallet.ts:302]: Would make a helper function for this and reuse everywhere 'isZkSyncChain()'

**Comment by 0xFirekeeper** [comment]
don't merge, webgl seems to call some api that triggers normal smart wallet flow, and react changes I need checked

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**